source/ControllableHero.hx

```diff
@@ -0,0 +1,80 @@
+package;
+
+import flixel.math.FlxPoint;
+import flixel.util.FlxColor;
+import flixel.FlxSprite;
+import flixel.FlxG;
+
+class ControllableHero extends FlxSprite
+{
+	var directionalAccel = new Array();
+
+	override public function new(x, y)
+	{
+		super(x, y);
+		maxVelocity = FlxPoint.get(200, 200);
+		directionalAccel.resize(4);
+		makeGraphic(15, 15, FlxColor.WHITE);
```

Comment:
Numbers like these 15s used in `makeGraphic` can be pretty mysterious, especially if we step away from the codebase for a while.

Are these both 15 because they were supposed to always match one another? Were they independently picked, and can be changed willy-nilly without issues? Are there any other spots in the code that depend on these staying unchanged?

It's good practice to take numbers like these and store them in class-level variables - static properties that can't be set to something new later is usually a good policy for const-type stuff.

Then we can give them names and make the intent of our code clearer - the player's starting_width should be 15, the player's starting_height should be 15.

Let's implement this sort of fix here and elsewhere in the project.

In some cases (like when you resize the array) there's less value to setting up some sort of constant - it would be nice to have some named variable to explain why we picked 4 (presumably because it's number of cardinal directions) but it's not the sort of data we'd like to have our `ControllableHero` define - the number of cardinal directions is something that we shouldn't be responsible for defining here.

I recommend using your judgement to determine which numbers have explainable values (or ones that may need reconfiguring in the future) that will benefit from being turned into class-level variables.

Reply:
But as a quick example, the `150` used in your update method to set the directional acceleration is a perfect candidate to be some kind of constant value - it'll be repeated and used in several places inside the class.